### ISSUE TITLE
more parameters included than before 2.0.0-alpha1 -> 2.0.0

### ISSUE THREAD

**Issue by gregjotau** [bigskysoftware/htmx#2661]
TL;DR: More parameters than before was included as request parameters than before, unchecked items was now included:

<img width="648" alt="image" src="https://github.com/bigskysoftware/htmx/assets/13047497/ee09b3b8-3397-48f7-a285-647062feb588">

```
<!--    <script src="https://unpkg.com/htmx.org@2.0.0" integrity="sha384-wS5l5IKJBvK6sPTKa2WZ1js3d947pvWXbPJ1OmWfEuxLgeHcEbjUUA5i9V5ZkpCw" crossorigin="anonymous"></script>-->

    <!-- potential bug where sku:qty pair of unchecked line item is sent to the server in request params in htmx 2.0.0 DO NOT UPGRADE BEFORE FIX -->
    <script src="https://unpkg.com/htmx.org@2.0.0-alpha1/dist/htmx.min.js"></script>
```

So I tried to upgrade to 2.0.0 from the alpha1 version. The only difference between working and not working version is in the import above the 2.0.0 version was used. 

I have the following shoelace form:

```
<form hx-include="#orderNumber, [checked] + .sku-quantity, #returnType"
      hx-indicator="#submitReturnRequestButton"
      hx-post="/iframes/submit-return-request"
      hx-swap="outerHTML"
      hx-target="#submitReturnRequestButton"
>
    <sl-alert open variant="primary">
        <b th:text="#{return.request.information}"/>
    </sl-alert>

    <div th:each="item : ${lineItems}">
        <sl-checkbox
                th:checked="${lineItems.size() == 1}"
                th:text="${item.nameSku}"
                th:value="${item.sku}"></sl-checkbox>
        <sl-select class="sku-quantity"
                   th:label="#{return.quantity}"
                   th:name="${item.sku}"
                   th:required="${lineItems.size() == 1 ? 'required' : null}"
                   th:value="${item.quantity == 1 ?  '1' : ''}"
        >
            <sl-option th:each="i : ${#numbers.sequence(1, item.quantity)}"
                       th:text="${i}"
                       th:value="${i}"></sl-option>
        </sl-select>
    </div>

    <sl-select id="returnType"
               name="returnType"
               required
               th:label="#{return.choose.refund.or.exchange}"
    >
        <sl-option th:text="#{exchange}" value="exchange"></sl-option>
        <sl-option th:text="#{refund}" value="refund"></sl-option>
    </sl-select>


    <sl-button id="submitReturnRequestButton"
               style="margin-bottom: 30px;"
               th:text="#{submit.return.request}"
               type="submit"
               variant="primary"
    >
    </sl-button>
</form>

```

And backend code:

```
@PostMapping("/iframes/submit-return-request")
    fun submitReturnRequest(
        @RequestParam orderNumber: String,
        @RequestParam returnType: String,
        @RequestParam allRequestParams: Map<String, String>,
        @RequestHeader(value = "X-Shopify-Store", required = true) shopifystore: String
    ): String {
        val userForStore = userRepository.findByShopifyStore(shopifystore)
        securityUtils.setCurrentUser(userForStore)
        val order = shopifyService.getOrder(orderNumber) ?: return "warning-order-not-found"
        val skuQtyMap = allRequestParams.filter {
            it.key != "orderNumber" && it.key != "returnType"
        }
```

Problem is the allRequestParams now included "all checkbox items" even unchecked ones.

**Comment by Telroshan** [comment]
Hey, I'm not sure I understand the situation here ; are you getting the unchecked checkboxes' name/value pairs on the backend?
Would you mind sharing a screenshot of your request's payload from your browser's dev tools' network tab?

I'm confused as I don't get the same result, see [this JSFiddle](https://jsfiddle.net/wnf4y7xz/) where I only get the checked checkbox's name/value pair as you would expect

Let me know!

**Comment by Telroshan** [comment]
I don't know shoelace at all, but are those values the name/value pairs of the checkboxes, or their what-seems-to-be-associated `select`s ? I'm asking as I don't see a `name` attribute defined for the checkboxes but do see one for the `select`s.
Is shoelace doing any values processing before htmx sends its request?

**Comment by Telroshan** [comment]
> But, I think the problem is that since the select has a value htmx still sends sends every name=value pair it finds.

Yep, and that's normal! This is why I asked about shoelace ; any input-type that has a name/value pair will be sent in a standard form, and htmx simply follows along the standard behavior on this.
It's likely that shoelace is doing some internal processing, that maybe doesn't work anymore with htmx 2. The only change I can think of is that we moved to `FormData` internally instead of the old non-standard plain object to represent the form values.
Though, we implemented this as a proxy object specifically for retro compatiblity so this might not be the culprit, but it's a suspect at least.

From what I see, sounds to me like that's an issue you'd want to solve at the shoelace level, as again the behavior itself is perfectly standard and valid 😕
